src/install.ts

```diff
@@ -1,21 +1,58 @@
 import * as core from "@actions/core"
 import * as tc from "@actions/tool-cache"
+import os from "os"
 import path from "path"
 import { run as setupGo } from "setup-go/lib/main"
+import util from "util"
 
 import { VersionConfig } from "./version"
 
+const defaultPlatform = "linux-amd64.tar.gz"
+
+const getAssetURL = (versionConfig: VersionConfig): string => {
+  let ext = "tar.gz"
+  let platform = os.platform().toString()
+  switch (platform) {
+    case "win32":
+      platform = "windows"
+      ext = "zip"
+      break
+  }
+  let arch = os.arch()
+  switch (arch) {
+    case "x64":
+      arch = "amd64"
+      break
+    case "x32":
+    case "ia32":
+      arch = "386"
+      break
+  }
+  const s = util.format("%s-%s.%s", platform, arch, ext)
+  if (s != defaultPlatform) {
+    return versionConfig.AssetURL.replace(defaultPlatform, s)
```

Comment:
understand that you are using AssetURL to cope with existing logic. I am thinking about using `TargetVersion` and then construct URL based on version. Then later, we can just remove redundant AssetURL in golangci-lint. 

Anyway, I am feel free to ignore this comment.

Reply:
to get rid of the `AssetURL` we need to came up with more complex logic of finding the latest version and latest patch version. Let's use it for now

Reply:
I see what you mean, yes, we can use `TargetVersion` to build the AssetURL

Reply:
done